api/server/router/image/backend.go

```diff
@@ -37,7 +37,7 @@ type importExportBackend interface {
 }
 
 type registryBackend interface {
-	PullImage(ctx context.Context, image, tag string, metaHeaders map[string][]string, authConfig *types.AuthConfig, outStream io.Writer) error
+	PullImage(ctx context.Context, image, tag string, metaHeaders map[string][]string, authConfig *types.AuthConfig, insecureRegistries []string, outStream io.Writer) error
```

Comment:
I'm not sure if this is relevant to the PR, but let's try to avoid argument bloat.